Themis/Views/Assessment/CorrectionSidebar/EditFeedback/EditFeedbackViewBase.swift

```diff
@@ -8,44 +8,73 @@
 import Foundation
 import SwiftUI
 
-struct EditFeedbackView: View {
-    @EnvironmentObject var avm: AssessmentViewModel
-    @EnvironmentObject var cvm: CodeEditorViewModel
+struct EditFeedbackViewBase: View {
+    @Binding var assessmentResult: AssessmentResult
+    @ObservedObject var cvm: CodeEditorViewModel
 
-    @State var feedbackText = ""
+    @Binding var feedback: AssessmentFeedback
+    @State var detailText = ""
     @State var score = 0.0
-    @Binding var showEditFeedback: Bool
+    @Binding var showSheet: Bool
 
     let maxScore = Double(10)
 
-    let feedback: AssessmentFeedback?
+    let title: String?
     let edit: Bool
     let type: FeedbackType
 
     var pickerRange: [Double] {
         Array(stride(from: -1 * maxScore, to: maxScore +  0.5, by: 0.5))
             .sorted { $0 > $1 }
     }
-    var title: String {
-        edit ? "Edit Feedback" : "Add Feedback"
+
+    func updateFeedback() {
+        feedback.updateFeedback(detailText: detailText, credits: score)
+        assessmentResult.updateFeedback(
+            id: feedback.id,
+            feedback: feedback
+        )
+    }
+
+    func createFeedback() {
+        var feedback = AssessmentFeedback(
+            detailText: detailText,
+            credits: score,
+            type: type,
+            file: cvm.selectedFile
+        )
+        let lines: NSRange? = cvm.selectedSectionParsed?.0
+        let columns: NSRange? = cvm.selectedSectionParsed?.1
+        feedback.buildLineDescription(lines: lines, columns: columns)
```

Comment:
This should rather be done in the feedback initialiser. Otherwise one might quickly forget this.

Reply:
With this I mean `feedback.buildLineDescription(lines: lines, columns: columns)`

Reply:
Changed that, it's nicer 👍